core/Plugin/ReportsProvider.php

```diff
@@ -40,9 +40,45 @@ public static function factory($module, $action)
             return null;
         }
 
-        $klassName = $listApiToReport[$api];
+        $reports = $listApiToReport[$api];
 
-        return new $klassName;
+        $report = self::findReportMatchingParameters($reports);
+        if (empty($report)) { // sanity check
+            return null;
+        }
+
+        $klassName = $report['class'];
+        return new $klassName();
+    }
+
+    private static function findReportMatchingParameters($reports)
+    {
+        $requestParams = $_GET + $_POST;
+
+        $match = null;
+        $matchParamCount = -1;
+
+        foreach ($reports as $report) {
+            $isMatchedWithThisReport = self::isReportParamsMatchingCurrentParams($report, $requestParams);
```

Comment:
From what I can see this may cause problems with various plugins that are based on entities and don't create a report instance for each entity, such as funnels. By default, eg `GetFunnelFlow` doesn't set parameters, but `$report->parameters` may be set when `configureReportMetadata` is called and I can imagine depending on the flow how methods are called that the logic might break some things. 

Same is happening in `GetDaysToConversion`  but they set `$this->parameters= array()` back in the `init` method which some other report classes don't do just yet. So there might be an easy fix for this by updating all our plugins...

Is this change needed for this PR?

Reply:
It was needed at one point w/ some other changes, but looks like it's not needed currently. Will remove it.